src/components/cc-addon-admin/cc-addon-admin.js

```diff
@@ -29,143 +29,182 @@ export class CcAddonAdmin extends LitElement {
-      saving: { type: Boolean },
+      state: { type: Object },
       _name: { type: String, state: true },
-      _skeleton: { type: Boolean, state: true },
-      _tags: { type: Array, state: true },
+      _tags: { type: String, state: true },
     };
   }
 
   constructor () {
     super();
 
-    /** @type {Addon|null} Sets the add-on details (name and tags). */
-    this.addon = null;
-
-    /** @type {boolean} Sets the error state on the component. */
-    this.error = false;
-
     /** @type {boolean} Hides the text about backups within the danger zone when set to `true` */
     this.noDangerZoneBackupText = false;
 
     /** @type {boolean} Hides the text about VM delay within the danger zone when set to `true` */
     this.noDangerZoneVmText = false;
 
-    /** @type {boolean} Enables the saving state */
-    this.saving = false;
+    /** @type {AddonAdminState} Sets the state of the component */
+    this.state = { type: 'loading' };
 
-    /** @type {string} */
-    this._name = '';
+    /** @type {string|null} Sets the value of the `name` input field */
+    this._name = null;
 
-    /** @type {boolean} */
-    this._skeleton = false;
+    /** @type {string[]|null} Sets the value of the `tags` input field */
+    this._tags = null;
+  }
 
-    /** @type {string[]} */
-    this._tags = [];
+  /** @private */
+  _onDeleteSubmit () {
+    dispatchCustomEvent(this, 'delete-addon');
   }
 
+  /**
+   * @param {{ detail: string }} event
+   * @private
+   */
   _onNameInput ({ detail: name }) {
     this._name = name;
   }
 
+  /** @private */
   _onNameSubmit () {
     dispatchCustomEvent(this, 'update-name', { name: this._name });
   }
 
+  /**
+   * @param {{ detail: string[] }} event
+   * @private
+   */
   _onTagsInput ({ detail: tags }) {
     this._tags = tags;
   }
 
+  /** @private */
   _onTagsSubmit () {
     dispatchCustomEvent(this, 'update-tags', { tags: this._tags });
   }
 
-  _onDeleteSubmit () {
-    dispatchCustomEvent(this, 'delete-addon');
-  }
-
-  _onDismissError () {
-    this.error = false;
+  /**
+   * @param {AddonAdminState} state
+   * @returns {state is (AddonAdminStateLoaded|AddonAdminStateLoading|AddonAdminStateSaving)}
+   * @private
+   */
+  _shouldRenderContent (state) {
+    const possibleStateTypes = ['loaded', 'loading', 'deleting', 'updatingName', 'updatingTags'];
+    return possibleStateTypes.includes(state.type);
   }
 
+  /**
+   * @param {CcAddonAdminPropertyValues} changedProperties
+   */
   willUpdate (changedProperties) {
+    if (changedProperties.has('state') && 'name' in this.state) {
+      this._name = this.state.name;
+    }
 
-    if (changedProperties.has('addon')) {
-      this._skeleton = (this.addon == null);
-      this._name = this._skeleton ? '' : this.addon.name;
-      this._tags = this._skeleton ? [] : this.addon.tags;
+    if (changedProperties.has('state') && 'tags' in this.state) {
+      this._tags = this.state.tags;
     }
```

Comment:
@hsablonniere we hesitated a lot with this one but here is our mindset regarding this.

At first, we decided to check for state types but as you can see below, it becomes very verbose:
```js
    if (changedProperties.has('state') && (this.state.type === 'loaded' || this.state.type === 'deleting' || this.state.type === 'updatingName' || this.state.type === 'updatingTags')) {
      this._name = this.state.name;
    }
```

We thought it was difficult to read and we didn't like it so we decided to go for the straightforward fix: TypeScript says it might cause errors because `name` may not exist? Then let's make sure `name` exists.
This is the only case where we used that.

Reply:
Agreed, and I checked, using `includes` will break inference.